codec-http2/src/main/java/io/netty/handler/codec/http2/Http2Connection.java

```diff
@@ -253,6 +253,30 @@
     }
 
     /**
+     * A key to be used for associating application-defined properties with streams within this connection.
+     *
+     * @param <K> The object type underlying the key.
+     * @param <V> The value type to be stored by this key.
+     */
+    interface PropertyKey<K, V> {
```

Comment:
> > Can we remove the V generic argument. Seems like it is not needed and is not used in this interface.

> The V parameter is useful for the get(Key) and remove(Key) methods so that they automatically cast the return value to the appropriate type.

Can't the generic argument `V` just be moved to the class (or method) that actually uses it?  Just seems a bit strange to have a generic argument on a type/interface but that type/interface doesn't use that generic type.